### PR TITLE
Session check standarization

### PR DESCRIPTION
**Pull request by Daegaladh**
* **Addressed Issue(s)**: N/A

* **Server Mode**: Both

* **Description of Pull Request**:

### CODE REVIEW

**Comment by secretdataz** [comment]
Please flag both relevant functions as inline while you're at it.

File: src/char/char_logif.cpp [hunk 1]
```diff
@@ -757,7 +757,7 @@ int chlogif_parse(int fd) {
 }
 
 TIMER_FUNC(chlogif_check_connect_logserver){
-	if (login_fd > 0 && session[login_fd] != NULL)
+	if (session_isValid(login_fd))
```
Comment (Lemongrass3110) [inline src/char/char_logif.cpp:760]: Replace with chlogif_isconnected()

Comment (Lemongrass3110) [inline src/char/char_logif.cpp:760]:
```suggestion
	if (chlogif_isconnected())
```

File: src/char/char_mapif.cpp [hunk 2]
```diff
@@ -72,7 +72,7 @@ int chmapif_sendallwos(int sfd, unsigned char *buf, unsigned int len){
  * @return : the number of map-serv the packet was sent to (O|1)
  */
 int chmapif_send(int fd, unsigned char *buf, unsigned int len){
-	if (fd >= 0) {
+	if (session_isValid(fd)) {
```
Comment (Lemongrass3110) [inline src/char/char_mapif.cpp:75]: You missed two cases above. In chmapif_sendall and chmapif_sendallwos. Please also change them.

File: src/map/clif.cpp [hunk 3]
```diff
@@ -9321,7 +9321,7 @@ void clif_GM_kick(struct map_session_data *sd, struct map_session_data *tsd)
 	if (sd == NULL)
 		tsd->state.keepshop = true;
 
-	if (fd > 0)
+	if (session_isValid(fd))
```
Comment (Lemongrass3110) [inline src/map/clif.cpp:9324]: Missing clif_authfail_fd

**Review by Lemongrass3110 — CHANGES_REQUESTED** [review]
I agree with @secretdataz that it would be a good improvement to inline these functions while we are it.

Looks good in general only missed a few references that I found by searching for "fd" variable references. Might be that we inherited some other creative variable names though. But we can always fix these later on.

Beside my in-code-comments I saw that you missed two files completely: intif.cpp and script.cpp
Please check them for references too.

**Comment by aleos89** [comment]
I think what @secretdataz and @Lemongrass3110 commented on are good as well.

**Review by aleos89 — APPROVED** [review]

**Review by Lemongrass3110 — APPROVED** [review]

**Pull request merged**